index.js

```diff
@@ -1,14 +1,21 @@
 const compiler = require('vue-template-compiler');
 const transpile = require('vue-template-es2015-compiler');
+const parse5 = require('parse5');
+const consolidate = require('consolidate');
+const spawnSync = require('child_process').spawnSync;
+const pathJoin = require('path').join;
 
 const exportsTarget = '((module.exports.default || module.exports).options || module.exports.default || module.exports)';
 
 module.exports = function ({ content, filename, sourceMap }) {
-  const startOfScript = content.indexOf('<script>') + 8;
-  const endOfScript = content.lastIndexOf('</script>');
-  if (startOfScript < 8 || endOfScript < 0 || endOfScript < startOfScript){
+  const documentFragment = parse5.parseFragment(content, {locationInfo: true});
```

Comment:
I've run this change with one of my projects that has 931 tests (across 130~ files). My total test time grows from an average of **25.573s** to **30.146s**. My templates are all written in html so that's without any actual calls to conslidate.js as well. 

I would quite like to see whether manually finding the template/script/lang cuts down this deficit as I want to keep performance as high as possible.